planning/behavior_path_planner/src/utilities.cpp

```diff
@@ -892,66 +924,25 @@ bool setGoal(
-      const double next_z = input.points.at(closest_seg_idx + 1).point.pose.position.z;
-      const double goal_z = std::abs(seg_dist) < 1e-6
-                              ? next_z
-                              : closest_z + (next_z - closest_z) * closest_to_goal_dist / seg_dist;
-
-      refined_goal.point.pose = goal;
-      refined_goal.point.pose.position.z = goal_z;
-      refined_goal.point.longitudinal_velocity_mps = 0.0;
-      refined_goal.lane_ids = input.points.back().lane_ids;
-    }
-
-    // calculate pre_goal
-    double roll{};
-    double pitch{};
-    double yaw{};
-    tf2::Quaternion tf2_quaternion(
-      goal.orientation.x, goal.orientation.y, goal.orientation.z, goal.orientation.w);
-    tf2::Matrix3x3 tf2_matrix(tf2_quaternion);
-    tf2_matrix.getRPY(roll, pitch, yaw);
+    const size_t closest_seg_idx_for_goal =
+      motion_utils::findNearestSegmentIndex(input.points, goal.position);
+    refined_goal.point.pose = goal;
+    refined_goal.point.pose.position.z =
+      calcInterpolatedZ(input, goal.position, closest_seg_idx_for_goal);
+    refined_goal.point.longitudinal_velocity_mps = 0.0;
 
     // calculate pre_refined_goal with interpolation
+    // NOTE: z and velocity are filled
     PathPointWithLaneId pre_refined_goal{};
-    pre_refined_goal.point.pose.position.x = goal.position.x - std::cos(yaw);
-    pre_refined_goal.point.pose.position.y = goal.position.y - std::sin(yaw);
-    pre_refined_goal.point.pose.orientation = goal.orientation;
-
-    {  // NOTE: interpolate z and velocity of pre_refined_goal
-      const size_t closest_seg_idx =
-        motion_utils::findNearestSegmentIndex(input.points, pre_refined_goal.point.pose.position);
-      const double closest_to_pre_goal_dist = motion_utils::calcSignedArcLength(
-        input.points, input.points.at(closest_seg_idx).point.pose.position,
-        pre_refined_goal.point.pose.position);
-      const double seg_dist = motion_utils::calcSignedArcLength(
-        input.points, closest_seg_idx,
-        closest_seg_idx + 1);  // TODO(murooka) implement calcSignedArcLength(points, idx, point)
-
-      const double closest_z = input.points.at(closest_seg_idx).point.pose.position.z;
-      const double next_z = input.points.at(closest_seg_idx + 1).point.pose.position.z;
-      pre_refined_goal.point.pose.position.z =
-        std::abs(seg_dist) < 1e-06
-          ? next_z
-          : closest_z + (next_z - closest_z) * closest_to_pre_goal_dist / seg_dist;
-
-      const double closest_vel = input.points.at(closest_seg_idx).point.longitudinal_velocity_mps;
-      const double next_vel = input.points.at(closest_seg_idx + 1).point.longitudinal_velocity_mps;
-      pre_refined_goal.point.longitudinal_velocity_mps =
-        std::abs(seg_dist) < 1e-06 ? next_vel : closest_vel;
-
-      pre_refined_goal.lane_ids = input.points.back().lane_ids;
-    }
+    pre_refined_goal.point.pose = tier4_autoware_utils::calcOffsetPose(goal, -1.0, 0.0, 0.0);
```

Comment:
minor thing, but please make a variable to represent the "goal to pre_goal" distance.
It would be like this.
```
constexpr auto goal_to_pre_goal_distance = -1,0;
pre_refined_goal.point.pose = tier4_autoware_utils::calcOffsetPose(goal, goal_to_pre_goal_distance, 0.0, 0.0);
```

Reply:
https://github.com/autowarefoundation/autoware.universe/pull/1626/commits/ceb9d754f0b6c5b1e4c9914ff78b98b58f4548d7
DONE